src/components/Form.js

```diff
@@ -1,17 +1,53 @@
-import React from 'react';
-
-const Form = () => (
-  <>
-    <h2>ADD NEW BOOK</h2>
-    <form>
-      <input type="text" />
-      <select name="categorie" id="categorie">
-        <option value="Action">Action</option>
-        <option value="Science Fiction">Science Fiction</option>
-        <option value="Economy">Economy</option>
-      </select>
-      <button type="submit">ADD BOOK</button>
-    </form>
-  </>
-);
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/books';
+
+const Form = () => {
+  const bookInfo = {
+    title: '',
+    category: '',
+  };
+
+  const dispatch = useDispatch();
+
+  const [book, setBook] = useState(bookInfo);
+
+  const { title, category } = book;
+
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setBook((prev) => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const submitBookToStore = (event) => {
+    event.preventDefault();
+    const newBook = {
+      id: uuidv4(),
+      title,
+      category,
+    };
+
+    dispatch(addBook(newBook));
+  };
+
+  return (
+    <>
+      <h2>ADD NEW BOOK</h2>
+      <form onSubmit={submitBookToStore}>
+        <input name="title" type="text" value={title} onChange={handleChange} />
+        <select name="category" id="categories" onChange={handleChange}>
+          <option value="none">Category</option>
+          <option value="Action">Action</option>
+          <option value="Science Fiction">Science Fiction</option>
+          <option value="Economy">Economy</option>
+        </select>
+        <button type="submit">ADD BOOK</button>
+      </form>
+    </>
```

Comment:
Sorry that the last CR didn't address this but is necessary to complete your project.

- [ ] In your React component responsible for adding new books set the data inputs in the local React state (set title and author).
* It's a requirement to also add an input to add an author and render the name of the author. 😉

Reply:
This issue is well resolved! ✔️